pkg/controller/disruption/disruption.go

```diff
@@ -204,16 +208,12 @@ func (dc *DisruptionController) getPodReplicaSet(pod *v1.Pod) (*controllerAndSca
 	return &controllerAndScale{rs.UID, *(rs.Spec.Replicas)}, nil
 }
 
-// getPodStatefulSet returns the statefulset managing the given pod.
-func (dc *DisruptionController) getPodStatefulSet(pod *v1.Pod) (*controllerAndScale, error) {
-	controllerRef := metav1.GetControllerOf(pod)
-	if controllerRef == nil {
-		return nil, nil
-	}
+// getPodStatefulSet returns the statefulset referenced by the provided controllerRef.
+func (dc *DisruptionController) getPodStatefulSet(controllerRef *metav1.OwnerReference, namespace string) (*controllerAndScale, error) {
 	if controllerRef.Kind != controllerKindSS.Kind {
```

Comment:
need to also make sure `schema.ParseGroupVersion(controllerRef.APIVersion).Group == "apps"`